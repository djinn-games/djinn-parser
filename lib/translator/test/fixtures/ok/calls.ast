{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "call",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "CallExpression",
          "callee": {
            "type": "Identifier",
            "name": "log",
            "line": 3
          },
          "args": [
            {
              "type": "Literal",
              "line": 3,
              "dataType": "str",
              "value": "hello world"
            }
          ],
          "line": 3
        },
        "line": 3
      }
    ],
    "line": 2
  },
  "line": 1
}