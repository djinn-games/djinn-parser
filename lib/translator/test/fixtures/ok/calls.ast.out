{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "CallExpression",
        "callee": {
            "type": "MemberExpression",
            "object": {
                "type": "Identifier",
                "name": "DJINN"
            },
            "property": {
                "type": "Identifier",
                "name": "log"
            }
        },
        "arguments": [
          {
            "type": "Literal",
            "value": "hello world",
            "raw": "\"hello world\""
          }
        ]
      }
    }
  ]
}
